### ISSUE TITLE
WebGL2 spec should specify what happens when querying default objects

### ISSUE THREAD

**Issue by greggman** [KhronosGroup/WebGL#2350]
In OpenGL 3.0 ES if you make a new context and query `VERTEX_ARRAY_BINDING` or `TRANSFORM_FEEDBACK_BINDING` you'll get `0` which is the default vertex array and default transform array.

     GLint currentVAO = -1; 
     glGetIntegeriv(GL_VERTEX_ARRAY_BINDING, &currentVAO);
     GLint currentTF = -1;
     glGetIntegeriv(GL_TRANSFORM_FEEDBACK_BINDING, &currentTF);

In WebGL2, in current implementations you get `null`. This is arguably inconsistent with OpenGL. In OpenGL you always get the same thing, an `GLint` which is an id, 0 happens to be the default and any other number is a user created resource.

To do the same in WebGL2 querying the default should return a `WebGLObject` that represents the default object in each case. That way it's consistent just like it is for OpenGL

Consider some code that wanted to augment VAOs for tracking

     var vao = gl.getParameter(gl.VERTEX_ARRAY_BINDING);
     if (!vao.trackingId) {
        vao.trackingId = nextTrackingId++;
     }

The code above would fail if the default is bound. That inconsistency makes it different than OpenGL where the default objects are not treated different.
 
I'm not saying that's the way it should be but ... if it's going to return `null` for the default that should be specced. As it is it seems like it would be okay for some implementations to return a `WebGLObject` for the default and others to not. Maybe it should also probably listed as a difference between OpenGL ES 3.0 and WebGL2 that the return value is inconsistent.

For Textures, since in WebGL1 and 2 there is no default texture and that's specified it's probably already indirectly clear that querying `TEXTURE_BINDING_2D` and `TEXTURE_BINDING_CUBE_MAP` etc would return null if no texture is bound.

**Comment by zhenyao** [comment]
At the time we didn't disallow default TF or VAO in concern that native contents may actually use it, where using the default textures are much less likely.

I don't think returning a WebGLObject is a good idea because it's confusing.  Now you bind that object or bind null have the same effects.  I guess we can clarify in the spec that querying returns null if default is bound.

**Comment by kenrussell** [comment]
I also think that returning a non-null WebGLObject for queries of the default VAO, transform feedback object, etc. is not a good idea because it'll be confusing. I am heavily inclined to tightly specify the current behavior and write conformance tests for it.

**Comment by greggman** [comment]
I agree the spec should probably say it returns 'null'. I'm just pointing out currently the spec doesn't say what happens so an implementation is free to do either.

It does make me wonder why default textures are not supported though. I never understood why they weren't in WebGL1 for that matter. I thought it was because it was a mistake and would be fixed in WebGL2
